epoch_publisher/src/client.rs

```diff
@@ -156,4 +208,45 @@ impl EpochPublisherClient {
         fbb.finish(fbb_root, None);
         fbb.finished_data()
     }
+
+    async fn background_loop(client: Arc<Self>, cancellation_token: CancellationToken) {
+        loop {
+            if cancellation_token.is_cancelled() {
+                return;
+            }
+
+            loop {
+                let expired_req_id = {
+                    let mut timeout_queue = client.timeout_queue.lock().await;
+                    if let Some(item) = timeout_queue.pop() {
+                        let (expiration_time, req_id) = item;
+                        if expiration_time < chrono::Utc::now() {
+                            req_id
+                        } else {
+                            break;
+                        }
+                    } else {
+                        break;
```

Comment:
possible improvement: you can sleep for the time till the next expiration OR cancellation using a `select`. Otherwise, it will be a busy loop.

Reply:
Added a todo. I think a fixed sleep for 10ms works for now, sleeping till next expiration would also need updating on new inserts which might (conceptually) have sooner expiration etc.